src/sidebar/audit-log.js

```diff
@@ -0,0 +1,129 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { CircularLoader } from '@dhis2/ui'
+import moment from 'moment'
+import PropTypes from 'prop-types'
+import React, { useEffect } from 'react'
+import styles from './audit-log.module.css'
+import LoadingError from './loading-error.js'
+import ToggleableUnit from './toggleable-unit.js'
+
+// TODO
+const query = {
+    auditLog: {
+        resource: 'auditLog',
+        id: ({ id }) => id,
+    },
+}
+
+// TODO: find better way to do this
+const renderMessage = ({ displayName, changeType, newValue, oldValue }) => {
+    switch (changeType) {
+        case 'UPDATE':
+            return i18n.t(
+                '{{displayName}} <span class="{{typeClass}}">updated</span> to {{newValue}} (was {{oldValue}})',
+                {
+                    displayName,
+                    typeClass: styles.entryType,
+                    newValue,
+                    oldValue,
+                }
+            )
+        case 'DELETE':
+            return i18n.t(
+                '{{displayName}} <span class="{{typeClass}}">deleted</span> (was {{oldValue}})',
+                {
+                    displayName,
+                    typeClass: styles.entryTypeDeleted,
+                    newValue,
+                    oldValue,
+                }
+            )
+        default:
+            return `${displayName} ${changeType}`
+    }
+}
+
+const Entry = ({ entry }) => {
+    const timeAgo = moment(entry.at).fromNow()
+
+    return (
+        <>
+            <span
+                className={styles.entryMessage}
+                dangerouslySetInnerHTML={{ __html: renderMessage(entry) }}
+            ></span>
+            <span className={styles.entryTimeAgo}>{timeAgo}</span>
+        </>
+    )
+}
```

Comment:
Couldn't find a better way to do this :grimacing:

Reply:
that's right :100:

Reply:
Guess this is an argument to look into supporting [react-i18next](https://react.i18next.com/latest/trans-component)

Reply:
I'm not sure of a better way than `dangerouslySetInnerHTML` to get the different styles within the text, but I'm wondering if `span` and `class` would get translated and become invalid HTML -- what do you think about:
```
i18n.t(
    '{{displayName}} {{htmlOpenTag}}updated{{htmlCloseTag}} to {{newValue}} (was {{oldValue}})',
    {
        displayName,
        htmlOpenTag: `<span class="${styles.entryType}">`,
        htmlCloseTag: '</span>'
        newValue,
        oldValue,
    }
)
```

Is there such thing as RTL html? The order of the tags matters I guess -- could be `htmlTagLeft` and `htmlTagRight` as keys maybe

Reply:
Yeah either solution relies on translators not accidentally messing with the HTML and I'm not sure which one is less error prone.

Reply:
I think we should just not use any html whatsoever until it's actually supported, unless this is really important.
If I remember correctly, that's the strategy we chose in the sms config app with @cooper-joe